### PR TITLE
Add test for complex abi

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: starknet_py/cairo/serialization/factory.py [hunk 1]
```diff
@@ -102,6 +102,16 @@ def serializer_for_payload(payload: Dict[str, CairoType]) -> PayloadSerializer:
     )
 
 
+def serializer_for_event(event: Abi.Event) -> PayloadSerializer:
```
Comment (Solpatium) [inline starknet_py/cairo/serialization/factory.py:105]: Added this as using payload serialiser is not obvious

Comment (cptartur) [reply starknet_py/cairo/serialization/factory.py:105]: Should this close #367 ?

**Review by cptartur — APPROVED** [review]

**Review by drknzz — APPROVED** [review]
Love the examples ❤️ 🦆
Maybe we could add a pointer in some struct?

What if someone had 3 pets?
🐸 🐶 🐦 ❓

**Review by war-in — APPROVED** [review]

**Review by drknzz — APPROVED** [review]

File: starknet_py/cairo/serialization/serialization_test.py [hunk 2]
```diff
@@ -19,21 +19,29 @@
 cat = {"name": encode_shortstring("Bob"), "species": encode_shortstring("cat")}
 cat_serialized = [cat["name"], cat["species"]]
 
-ceo = {"name": encode_shortstring("Scrooge McDuck"), "pet": dog}
-ceo_serialized = [ceo["name"], *dog_serialized]
+bird = {
+    "name": encode_shortstring("Yacub"),
+    "species": encode_shortstring("Parus major"),
```
Comment (war-in) [inline starknet_py/cairo/serialization/serialization_test.py:24]: 🐦 
image

**Review by war-in — APPROVED** [review]

**Pull request merged**